Sources/Core/AWSSDKIdentity/Sources/AWSSDKIdentity/S3Express/S3ExpressCreateSessionClient.swift

```diff
@@ -0,0 +1,15 @@
+//
+// Copyright Amazon.com Inc. or its affiliates.
+// All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import protocol ClientRuntime.DefaultClientConfiguration
+
+public protocol S3ExpressCreateSessionClient {
+    func createSession(
+        clientConfig: DefaultClientConfiguration,
+        bucket: String
+    ) async throws -> S3ExpressIdentity
+}
```

Comment:
This protocol provides an interface for the S3 Express identity resolver to call S3 `CreateSession` to get credentials, without depending on the code-generated S3 client.

A SwiftIntegration on the S3 service provides conformance with this protocol.